// Copyright 2013 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Author: gee@google.com (Adam Gee)
syntax = "proto2";

option optimize_for = LITE_RUNTIME;
/*


*/

package net_instaweb;

// Rewriter application status codes.
message RewriterApplication {
  enum Status {
    UNKNOWN_STATUS = 0;
    APPLIED_OK = 1;
    NOT_APPLIED = 2;
    PROPERTY_NOT_FOUND = 3;
    INPUT_URL_INVALID = 4;
    REPLACE_FAILED = 5;
  }
}

// Status of rewriter at the HTML level.  This would include failure reasons
// such as the rewriter being disabled, the url being blacklisted, or missing
// the property cache (for those rewriters which rely on properties).
message RewriterHtmlApplication {
  enum Status {
    UNKNOWN_STATUS = 0;
    ACTIVE = 1;
    DISABLED = 2;
    PROPERTY_CACHE_MISS = 3;
    USER_AGENT_NOT_SUPPORTED = 4;
  }
}
